pandas/io/excel/_base.py

```diff
@@ -346,33 +353,33 @@
 @Appender(_read_excel_doc)
 def read_excel(
     io,
-    sheet_name=0,
-    header=0,
+    sheet_name: str | int | list[int] | list[str] | None = 0,
+    header: int | Sequence[int] | None = 0,
     names=None,
-    index_col=None,
+    index_col: int | Sequence[int] | None = None,
     usecols=None,
```

Comment:
Based on the documentation, it could be: `usecols: None | str | list[int] | list[str] | Callable[[object], bool]` (could probably narrow the argument type for the callable).

Reply:
I have to look into this a bit more, We might be able to use Sequence instead of list here, hence did not want to type the public api yet